crud/src/main/java/com/redhat/lightblue/mediator/DefaultMetadataResolver.java

```diff
@@ -180,25 +191,81 @@ private void addFieldRoles(Set<String> roles, FieldTreeNode node) {
         }
     }
 
-    private EntityMetadata getMetadata(String entityName,
-                                       String version) {
-        EntityMetadata emd = metadataMap.get(entityName);
-        if (emd != null) {
-            if (!emd.getVersion().getValue().equals(version)) {
-                throw Error.get(CrudConstants.ERR_METADATA_APPEARS_TWICE, entityName + ":" + version
-                        + " and " + emd.getVersion().getValue());
+    private void initMetadataMap(EntityMetadata emd, String entityName, String rootRequestVersion) {
+        // store root entity metadata
+        metadataMap.put(entityName, emd);
+        Set<String> deferred = new HashSet<>();
+        FieldCursor cursor = emd.getFieldCursor();
+        while (cursor.next()) {
+            FieldTreeNode node = cursor.getCurrentNode();
+            if (node instanceof ReferenceField) {
+                ReferenceField ref = (ReferenceField) node;
+                String name = ref.getEntityName();
+                String ver = ref.getVersionValue();
+                EntityMetadata refMd;
+                if (ver == emd.getEntityInfo().getDefaultVersion()) {
+                    refMd = metadataMap.get(entityName);
+                } else {
+                    refMd = md.getEntityMetadata(name, ver);
```

Comment:
If ver!=null, then association requires a definite version. If a different version of the entity is already used, give error. If default version of the entity is referenced before, then that default version should be this version.